plugins/woocommerce/includes/tracks/class-wc-tracks-event.php

```diff
@@ -55,7 +55,7 @@ public function __construct( $event ) {
 	 * @return bool Always returns true.
 	 */
 	public function record() {
-		if ( wp_doing_ajax() || Constants::is_true( 'REST_REQUEST' ) ) {
+		if ( wp_doing_ajax() || Constants::is_true( 'REST_REQUEST' ) || wp_doing_cron() ) {
```

Comment:
I'm a bit wary of this change, would it have any unintended side effects during a cronjob run? I'm unsure how to test it wholly though.

Reply:
That's a good point. I did not see any test failures, but it would be nice if we could get a feedback from @woocommerce/proton 

Hey @woocommerce/proton ✋ Do you have any concerns with this change?

Reply:
No concerns but, in relation to the goals of the PR, I wonder if we should do:

```suggestion
		if ( wp_doing_ajax() || Constants::is_true( 'REST_REQUEST' ) || Constants::is_true( 'WP_CLI' ) || wp_doing_cron() ) {
```

My thinking being that there are essentially three different contexts in which Action Scheduler runs: 

- Async requests (already covered by `wp_doing_ajax()`)
- WP CRON (now covered by `wp_doing_cron()`, which you just added)
- WP CLI (which my suggestion would cover)

Sometimes, sites only run Action Scheduler via WP CLI and block it from running in the other contexts.

Reply:
Thank you for the change! LGTM!